src/main/java/bio/terra/app/configuration/ApplicationConfiguration.java

```diff
@@ -434,7 +434,7 @@ public String tdrServiceAccountEmail() throws IOException {
     if (defaultCredentials instanceof ServiceAccountCredentials) {
       return ((ServiceAccountCredentials) defaultCredentials).getClientEmail();
     }
-    return null;
+    return "";
```

Comment:
I assume this is to make running tests easier? If so maybe it would be better to add this to the test configurations instead: `TestApplicationConfiguration` and `EmbeddedDataRepoTestConfiguration`. I think we want non-test startup to fail with a NPE instead of continuing to run, and changing this code would remove that feature.

Reply:
It doesn't really NPE...it just doesn't find the bean

Reply:
OK, but if this is for tests, changing the test configuration would be a better solution than changing the non-test configuration.

Reply:
I also want to understand this better.  I've been frustrated by how the need for this bean has gotten in the way of unrelated unit tests in the past so I'm excited for a change, but want to make sure that non-test set-up is still working as expected.

Reply:
This bean is used to give the user the TDR service account email so that they know what to grant permissions to if it's not a per-dataset dedicated service account.  All pods have the default service account google credentials so that code only ever gets reached when running locally if you've `gcloud auth`'ed with, say, your dev user. This is why I like optionals...

Reply:
Why is this change here? Is it to better support tests, or to fix a different problem, such as running locally when no google auth is present?

> This is why I like optionals

IMO option doesn't help here, but if you want to, I believe spring supports autowiring for `Optional<>` types already. That is, if you inject a type into an optional, and the type can't be created, spring will create an empty optional.

Reply:
Huh interesting idea for the `@Nullable` annotation option.  I'll try that.  Honestly, that's all I'm trying to do.  nulls don't have types so Spring by default didn't know how to find the bean.

Reply:
The autowired nullable/optional behavior is described here: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html. It was new in spring 5.0.